### PR TITLE
adds fetchUserChannels, updates fetchReactionsForCast types parameter, and depreciates limit notes for fetchTrendingChannels

### PR DESCRIPTION
**Pull request by MrKevinOConnell**
New Method

User

fetchUserChannels -> https://docs.neynar.com/reference/user-channels

Updates Existing Methods

Reaction

fetchReactionsForCast -> https://docs.neynar.com/reference/reactions-cast
- types is now an array of ReactionsType, we join the array into a string at the end.

misc

Feed

fetchTrendingChannels -> https://docs.neynar.com/reference/trending-channels fixed limit

### CODE REVIEW

File: src/neynar-api/v2/client.ts [hunk 1]
```diff
@@ -1803,7 +1840,7 @@ export class NeynarV2APIClient {
    * @param {'1d' | '7d' | '30d'} [timeWindow] - The time window for trending analysis. Options are '1d' (one day),
    *   '7d' (seven days), or '30d' (thirty days).
    * @param {Object} [options] - Optional parameters to tailor the request.
-   * @param {number} [options.limit=25] - The number of channel details to fetch per request. Defaults to 25, with a maximum allowable value of 100.
+   * @param {number} [options.limit=10] - The number of channel details to fetch per request. Defaults to 10, with a maximum allowable value of 25.
```
Comment (Shreyaschorge) [inline src/neynar-api/v2/client.ts:1843]: Default is 25

File: src/neynar-api/neynar-api-client.ts [hunk 2]
```diff
@@ -2062,7 +2091,7 @@ export class NeynarAPIClient {
    * @param {'1d' | '7d' | '30d'} [timeWindow] - The time window for trending analysis. Options are '1d' (one day),
    *   '7d' (seven days), or '30d' (thirty days).
    * @param {Object} [options] - Optional parameters to tailor the request.
-   * @param {number} [options.limit=25] - The number of channel details to fetch per request. Defaults to 25, with a maximum allowable value of 100.
+   * @param {number} [options.limit=10] - The number of channel details to fetch per request. Defaults to 10, with a maximum allowable value of 25.
```
Comment (Shreyaschorge) [inline src/neynar-api/neynar-api-client.ts:2094]: Default is 25

File: src/neynar-api/neynar-api-client.ts [hunk 3]
```diff
@@ -1839,7 +1868,7 @@ export class NeynarAPIClient {
    */
   public async fetchReactionsForCast(
     hash: string,
-    types: ReactionsType,
+    types: ReactionsType[],
```
Comment (Shreyaschorge) [inline src/neynar-api/neynar-api-client.ts:1871]: This will be a breaking change for existing users

**Review by Shreyaschorge — CHANGES_REQUESTED** [review]

File: src/neynar-api/v2/client.ts (hunk 1 above)
Comment (MrKevinOConnell) [reply src/neynar-api/v2/client.ts:1843]: see here @Shreyaschorge https://github.com/neynarxyz/nodejs-sdk/pull/90/files#r1559907457

File: src/neynar-api/neynar-api-client.ts (hunk 3 above)
Comment (MrKevinOConnell) [reply src/neynar-api/neynar-api-client.ts]: reverted 👍🏻

File: src/neynar-api/v2/openapi-farcaster/apis/channel-api.ts [hunk 4]
```diff
@@ -494,7 +494,7 @@ export const ChannelApiFp = function(configuration?: Configuration) {
          * @summary Retrieve trending channels based on activity
          * @param {string} apiKey API key required for authentication.
          * @param {'1d' | '7d' | '30d'} [timeWindow] 
-         * @param {number} [limit] Number of results to retrieve (default 25, max 100)
+         * @param {number} [limit] Number of results to retrieve (default 10, max 25)
```
[the remark this answers is not in the document]
Comment (MrKevinOConnell) [reply src/neynar-api/v2/openapi-farcaster/apis/channel-api.ts:497]: https://docs.neynar.com/reference/trending-channels

[the remark this answers is not in the document]
Comment (Shreyaschorge) [reply src/neynar-api/v2/openapi-farcaster/apis/channel-api.ts:497]: @MrKevinOConnell Default here is incorrect, need to update OAS. 
Please checkout `route_config.json` in api

File: src/neynar-api/neynar-api-client.ts [hunk 5]
```diff
@@ -1814,7 +1844,7 @@ export class NeynarAPIClient {
    * the reactions associated with a cast, specified by the cast hash.
    *
    * @param {string} hash - The hash of the cast whose reactions are being fetched.
-   * @param {ReactionsType} types - The type of reaction to fetch (likes, recasts, or all).
+   * @param {ReactionsType[]} types - Commas separated string of types of reactions to fetch (likes, recasts, or all).
```
Comment (Shreyaschorge) [inline src/neynar-api/neynar-api-client.ts:1847]: @MrKevinOConnell Please update the JS Docs

**Review by Shreyaschorge — CHANGES_REQUESTED** [review]
Minor Changes the rest looks good

**Comment by manan19** [comment]
lgtm

**Review by Shreyaschorge — APPROVED** [review]
LGTM

**Pull request merged**